File: services/student-service/src/main/java/com/fuji/student_service/services/StudentService.java
package com.fuji.student_service.services;

import com.fuji.student_service.DTO.StudentRequest;
import com.fuji.student_service.DTO.StudentResponse;
import com.fuji.student_service.models.Ecolage;
import com.fuji.student_service.models.Note;
import com.fuji.student_service.models.Report;

import java.util.List;
import java.util.Map;

public interface StudentService {
    StudentResponse create(StudentRequest request);
    StudentResponse update(StudentRequest request);
    StudentResponse get(String id);
    StudentResponse getByMatricule(String matricule);
    List<StudentResponse> getAll();
    List<StudentResponse> getAllByIds(List<String> ids);
    StudentResponse addNotes(Note note);
    StudentResponse paymentEcolage(Ecolage ecolage);
    StudentResponse addReport(Report report);
    StudentResponse delete(String matricule);
}

File: front-ui/school-flow-ui/src/app/services/student.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { environment } from '../../environments/environment.development';
import { Student } from '../models/student.model';
import { Observable } from 'rxjs';
import { Note } from '../models/note.model';
import { Ecolage } from '../models/ecolage.model';

@Injectable({
  providedIn: 'root'
})
export class StudentService {
  httpClient= inject(HttpClient);

  url: string= environment.API_URL+'/STUDENT-SERVICE/api/student';

  constructor() { }

  create(student: Student): Observable<any> {
    return this.httpClient.post(this.url+'/create', student);
  }

  update(student: Student): Observable<any> {
    return this.httpClient.put(this.url+'/update', student);
  }

  get(matricule: string): Observable<any> {
    return this.httpClient.get(this.url+'/get/'+matricule);
  }

  getAll(): Observable<any> {
    return this.httpClient.get(this.url+'/all');
  }

  getByClassID(classID: string): Observable<any> {
    return this.httpClient.get(this.url+'/get-by-classID/'+classID);
  }

  addNote(note: Note): Observable<any> {
    return this.httpClient.patch(this.url+'/add-note', note);
  }

  paymentEcolage(ecolage: Ecolage): Observable<any> {
    return this.httpClient.patch(this.url+'/payment-ecolage', ecolage);
  }

  addReport(report: Report): Observable<any> {
    return this.httpClient.patch(this.url+'/add-report', report);
  }

  delete(matricule: string): Observable<any> {
    return this.httpClient.delete(this.url+'/delete/'+matricule);
  }
}
